crates/ruff/src/directives.rs

```diff
@@ -217,6 +219,154 @@ fn extract_isort_directives(lxr: &[LexResult], locator: &Locator) -> IsortDirect
     }
 }
 
+/// A comment that contains a [`TodoDirective`]
+pub struct TodoComment<'a> {
```

Comment:
Nit: Can we make the struct `pub(crate)`?

Reply:
Is there a reason we're migrating towards more crate-private structs? Seems like I've seen way more of them recently

Reply:
@MichaReiser wanted to bump this - I think it may be a blocker?

Reply:
Sorry. I missed this notification. 

If we can make the rule `pub(crate)` too, then that's what I would do. 

> Is there a reason we're migrating towards more crate-private structs? Feels like I've seen way more of them recently.

Restricting the visibility has the advantage that rust's dead code analysis works (disabled for all `pub` items because it may not know all downstream crates). Keeping the visibility low also has the advantage that it is easier to reasons about the code: Ah, *this* is only used in this crate, I can change this freely vs. Oh, *this* is used publicly, do I know all call sites? Which crates are using it?

Reply:
Sorry. I missed this notification. Thanks for the ping.

If we can make the rule `pub(crate)` too, then that's what I would do. 

> Is there a reason we're migrating towards more crate-private structs? Feels like I've seen way more of them recently.

Restricting the visibility has the advantage that rust's dead code analysis works (disabled for all `pub` items because it may not know all downstream crates). Keeping the visibility low also has the advantage that it is easier to reasons about the code: Ah, *this* is only used in this crate, I can change this freely vs. Oh, *this* is used publicly, do I know all call sites? Which crates are using it?

Reply:
Gotcha, thank you! Done :)